src/cmd-buildextend-metal

```diff
@@ -270,20 +270,28 @@ runvm "${target_drive[@]}" -- \
             --rootfs "${rootfs_type}" \
             "${disk_args[@]}"
 /usr/lib/coreos-assembler/finalize-artifact "${path}.tmp" "${path}"
-echo "{}" > tmp/vm-iso-checksum.json
 
-# there's probably a jq one-liner for this...
-python3 -c "
+# calculate this ahead of time to reduce critical section timespan
+sha256=$(sha256sum_str < "${img}")
+
+# and now the crucial bits
+# shellcheck disable=SC2094
+(echo "Waiting for lock..."
+ flock 9
```

Comment:
So today we do the "create new, `rename()`" dance.  Is this safe to do with that pattern?  It's not clear to me that it is.

I think it's simpler to use a separate lock file like `tmp/meta-json.lock`.

Reply:
> So today we do the "create new, rename()" dance. Is this safe to do with that pattern? It's not clear to me that it is.

As long as the critical section always finishes with `rename()` to the final `meta.json`, there's no race possible. Either another process tries locking the same inode and it blocks, or it locks the new inode, which is the last step we do anyway.

> I think it's simpler to use a separate lock file like `tmp/meta-json.lock`.

Hmm, yeah maybe... I do like the explicitness of locking `meta.json` directly. And it's also more immediately obvious that it's the same directory targeted between any process that needs to update `meta.json` since that's the source of truth. Whereas, even though `$workdir/tmp` is shared today, I could imagine us making that more flexible in the future for some reason (similar to what motivated https://github.com/coreos/coreos-assembler/pull/1485).

Reply:
That said, I'm totally fine using a separate lockfile too (though maybe in the build dir itself, e.g. `$builddir/meta.json.lock`? We'll have to make sure that it doesn't accidentally get uploaded in `buildupload`).

Reply:
One thing to keep in mind is that RHCOS pipeline is using NFS for `builds/` and that gets into http://0pointer.de/blog/projects/locking.html